### PR TITLE
Fixed NaN loss when given a mask with all zeros

### PR DESCRIPTION
**Pull request by CyberZHG**
### Summary

Sometimes when operates on the masks of sequential/time-series data, there will be a small chance that all values in a mask are zeros (e.g. only several positions are chosen at random with a small rate for loss calculation). The epsilon term was added to avoid NaN loss.

### Related Issues

N/A

### PR Overview

- [x] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: tests/test_loss_masking.py [hunk 1]
```diff
@@ -21,6 +21,11 @@ def test_masking():
     loss = model.train_on_batch(x, y)
     assert loss == 0
 
+    x = np.array([[[0], [0]],
```
Comment (gabrieldemarmiesse) [inline tests/test_loss_masking.py]: Could it be a separate test? Tests should be as small as possible and modular. You should make a helper function to create the model and avoid code duplication.

something like `test_mask_is_all_zeros()`.

Comment (CyberZHG) [reply tests/test_loss_masking.py]: @gabrieldemarmiesse I've updated the tests.
